policies/legacy-migration.md

```diff
@@ -0,0 +1,39 @@
+# Legacy Provider Policy
+## Purpose
+The Legacy Provider exists to create an opt-in availability mechanism for
+algorithms that, for various reasons, should have their use discouraged.  These
+reasons include, but are not limited to:
+* Discovered security issues leaving the algorithm in question unsafe for
+  general use
+* Lack of popular use (i.e. balancing code size vs consumption frequency)
+
+OpenSSL recognizes that consumption of these algorithms may continue to be
+required by consuming applications after the conditions above have been
+recognized.  The Legacy provider exists to provide a mechanism for such
+applications to continue to access these algorithms while allowing applications
+that don't require them to inadvertently continue to use them.
+
+## Constraints on moving an algorithm to the legacy provider
+1) Migration of an algorithm to the legacy provider must occur on a semantically
+versioned major release boundary.  Once a major release includes a given
+algorithm in a given provider, it must remain there for every minor release in
+that major stream
+
+2) Prior to migration, the migration must be announced for at least 1
+semantically versioned patch level release (see announcement mechanism below)
```

Comment:
my thinking was that in the event of a security issue, this would allow us to react faster. i.e. generally we can strive for a minor version worth of warning, but we reserve the right to migrate quickly in the event that we have to.  But if you feel strongly about it, I'm happy to change it.

Reply:
I do not think a security issue would trigger an immediate migration. I could see that a major algorithm weakness would definitely trigger some action such as deprecating an algorithm from the default provider but as we cannot really remove it in a minor version IMO minor release boundary for deprecations is also ok. Please note that we do not (and cannot) have fixed schedule for patch releases as security issues potentially trigger an immediate patch release.